UefiPayloadPkg/UefiPayloadPkg.dsc

```diff
@@ -406,6 +440,21 @@
   BaseCryptLib|CryptoPkg/Library/BaseCryptLib/SmmCryptLib.inf
 !endif
 
+!if $(MEASURED_BOOT_ENABLE) == TRUE && $(SMM_SUPPORT) == TRUE
+ Tcg2PhysicalPresenceLib|SecurityPkg/Library/SmmTcg2PhysicalPresenceLib/SmmTcg2PhysicalPresenceLib.inf
+!endif
+
+
+[LibraryClasses.common.DXE_DRIVER, LibraryClasses.common.UEFI_DRIVER, LibraryClasses.common.UEFI_APPLICATION]
+!if $(SECURE_BOOT_ENABLE) == TRUE
+  BaseCryptLib|CryptoPkg/Library/BaseCryptLib/BaseCryptLib.inf
+!endif
+
+[LibraryClasses.common.DXE_RUNTIME_DRIVER]
+!if $(SECURE_BOOT_ENABLE) == TRUE
+  BaseCryptLib|CryptoPkg/Library/BaseCryptLib/RuntimeCryptLib.inf

```

Comment:
The macro "CRYPTO_PROTOCOL_SUPPORT" can control it why we need to override it here.

Reply:
It's probably irrelevant now that it seems https://github.com/tianocore/edk2/pull/6222 is going in, but the reason is that CryptoPkg/Library/BaseCryptLibOnProtocolPpi/DxeCryptLib is not valid for DXE_RUNTIME_DRIVER. That's because it calls into CryptoDxe, which is a regular DXE driver, so it's being freed at ExitBootServices()